flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala

```diff
@@ -2957,6 +2957,136 @@ class TableEnvironmentTest {
     checkData(expectedResult1.iterator(), modelResult2.collect())
   }
 
+  @Test
```

Comment:
As AI endpoints can involve complex objects. I suggest adding some tests with more complex objects, arrays of arrays , arrays of objects, nested objects and arrays of nested objects.

Reply:
Sure I'll add more types